unittests/XmlParser/XmlParser.cpp

```diff
@@ -189,6 +189,7 @@ private slots:
     void brokenXML()
     {
         QFile file(":/broken.xml" );
+        QVERIFY( file.open( QIODevice::ReadOnly ) );
```

Comment:
Clearly it would be better to open the file inside loadFromXML, if !isOpen(). This would preserve source code compatibility for users of the KDReports::Report class.